api/api.go

```diff
@@ -217,6 +223,21 @@ func (a *API) Store(ctx context.Context, data io.Reader, size int64, toEncrypt b
 // Resolve a name into a content-addressed hash
 // where address could be an ENS name, or a content addressed hash
 func (a *API) Resolve(ctx context.Context, address string) (storage.Address, error) {
+	// if address is .rsk, resolve it with RNS resolver
+	eTLD, _ := publicsuffix.PublicSuffix(address)
```

Comment:
I reject this. You already have a `Resolver` interface that does exactly what the `RNSResolver` does. The resolver functionality should accommodate for different providers you could resolve through. Adding another 30 lines of code here for each provider seems overly redundant. I don't see why this method should change _at all_

Reply:
In this case, RNS resolver should need to implement ResolveValidator to be added to MultiResolver, which I do not think that is needed. At least, for now. This is very localised code that can be generalised later if needed.

Reply:
Unfortunately, the current implementation of the RNS contracts does not behave _exactly_ like the regular (ENS) resolver contracts, which is a problem we are trying to solve with the library.

If it did, it would be enough to just initialize `swarm` with something like:
```shell
--ens-api rsk:RNSresolverAddress@endpoint
```
(although it would probably be best to change the name of the flag as @acud suggested)

We've tried this out, and it seems that the response from the contract has a different format than expected. So it's not homogeneous with the rest of the resolvers that you could add/refer to in the flags. 

In the end we are letting the library—which understands these contracts despite their differences with the ENS ones—deal with resolving these addresses.